deid/dicom/parser.py

```diff
@@ -160,8 +160,9 @@ def get_nested_field(self, field, return_parent=False):
             # Otherwise it's an index into a sequence
             else:
 
```

Comment:
Here's an explanation of the change below:

If we fall into the else, the uid is a sequence.  We don't want to check for it's presence in a dictionary, but instead want to check to ensure that the uid (index) is within the bounds of the array of items in the SQ DataElements value which is a Sequence - an iterable.  If the index is outside the bounds we return None as the parent.  If the index is within the bounds of the array, we want to return the DataElement at that index (the specific occurrence of the 
sequence element) for update.